### ISSUE TITLE
beartype does not work with NamedTuple

### ISSUE THREAD

**Issue by murphyk** [beartype/beartype#181]
The following fails with beartype 0.11.0, and python 3.8, but works with typeguard.

```

from typing import NamedTuple

from beartype import beartype as typechecker
#from typeguard import typechecked as typechecker

@typechecker
class Foo(NamedTuple):
    mean: float

f  = Foo(42.0) # correct - but fails with beartype
g = Foo("bar") # correctly raises error with typeguard
```

Beartype complains `KeyError: 'namedtuple_Foo' on line `f=Foo`.

@patrick-kidger  @slinderman

**Comment by leycec** [comment]
**Gah!** This cannot be... *but it is.*

Clearly, my permissive development style of only testing core functionality when users justifiably complain about critical breakage has come back to haunt us all. Let's see what I can grind out over the next few days to resolve this for everyone. :face_exhaling: 

Arise, @beartype! Arise from your broken slumber.

**Comment by leycec** [comment]
**Aha!** This is happening because @beartype fully supports [PEP 563](https://peps.python.org/pep-0563/); `typeguard` doesn't and thus slow-dodges the spray of bullets here. @beartype: "Why aren't you as lazy as `typeguard`?"

To fully resolve PEP 563, @beartype necessarily inspects the standard `__module__` dunder attribute of each method of the decorated class to be type-checked. Oddly, methods inherited from the `typing.NamedTuple` class declare their `__module__` to be a fake placeholder module that doesn't actually exist called `"namedtuple_Foo"`. Since that module doesn't actually exist, looking that module up in the standard `sys.modules` dictionaries fails with the non-human-readable exception you note.

In other words, `typing.NamedTuple` lies and @beartype pays the price. The solution is for @beartype to stop trusting the `__module__` attribute. Paranoia, fear, and shock are our three primary weapons in the war against the `typing` module.

Well, isn't *that* horrifying. I am now regurgitating all over myself. :vomiting_face:

**Comment by leycec** [comment]
[**"Probabilistic Machine Learning."**](https://github.com/probml) Yes, I can see why @beartype would be of interest to you. :grin:

By which I meant to say... **resolved by 6cbfb59!** This hot fix will be published shortly with our upcoming `beartype 0.12.0` release.

In fact, your jaw-dropping example has awakened me to an obvious realization. Thanks to class decoration support recently added with `beartype 0.11.0`, @beartype can now deeply type-check *all* `typing.NamedTuple` type hints just by internally decorating those hints with @beartype. What I'm hinting at <sup>...*get it?*</sup> is that `beartype 0.12.0` will also deeply type-check *all* `typing.NamedTuple` type hints for you. You can still explicitly decorate your `typing.NamedTuple` subclasses with @beartype if you like, but that won't be strictly necessary anymore under `beartype ≥ 0.12.0`. `\o/`

Thanks again for the excellent writeup, @murphyk. This is why one of us is a well-regarded author.

**Comment by slinderman** [comment]
Amazing! Thanks for the quick fix and the detailed (and very entertaining) explanation!
